test/tests/performance/pbs_equiv_classes_perf.py

```diff
@@ -108,12 +112,159 @@ def test_basic(self):
 
         # wait for cycle to finish
         m = self.scheduler.log_match("Leaving Scheduling Cycle", starttime=t,
-                                     max_attempts=120)
+                                     max_attempts=200)
         self.assertTrue(m)
 
         c = self.scheduler.cycles(lastN=1)[0]
         cycle2_time = c.end - c.start
 
         self.logger.info('Cycle 1: %d Cycle 2: %d Cycle time difference: %d' %
                          (cycle1_time, cycle2_time, cycle1_time - cycle2_time))
-        self.assertTrue(cycle1_time > cycle2_time)
+        self.assertTrue(cycle1_time >= cycle2_time)
+
+    def test_server_queue_limit(self):
+        """
+        Test the performance with hard and soft limits
+        on resources
+        """
```

Comment:
I don't see this test as a performance test.  To see performance you will need to show the scheduler being slow and then it being faster using the equiv class feature.  Right now you submit 8 different classes and run a cycle.  You then run another cycle without changing anything.  These two cycles should take the same amount of time.  I see two ways you can go with this test.  You could do something similar to what test_basic() does.  Submit N different classes and run one cycle.  You can then break those jobs up into your 8 classes and rerun your cycle.  Show the difference in time.  The second thing you could do is to move this to the functional tests.  It is useful there because there isn't really a test that tests all the different ways to differentiate classes.  The tests I wrote basically would test one of them at a time.  This one tests 8 different ones.